app/validators/email_address_validator.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# NOTE: I initially wrote this as `EmailValidator` but it ended up clashing
+# with an indirect dependency of ours, `validate_email`, which, turns out,
+# has the same approach as we do, but with an extra check disallowing
+# single-label domains. Decided to not switch to `validate_email` because
+# we do want to allow at least `localhost`.
```

Comment:
The rationale in this comment might be more useful as added model spec on user, or adding validator specs for this new validator?

Separately - how does the devise validation differ/add/interact with this?

Reply:
Added a test for this. Devise validation has an extremely coarse regexp that allows things that vaguely look like an email address but aren't. This approach uses the same parser that is actually used by Rails when sending mails, so that we are sure that the code agrees on what an email address is, before we actually try to send it.